src/query/expression/src/kernels/concat.rs

```diff
@@ -98,19 +98,24 @@ impl Column {
     ) -> Result<Column> {
         let (_, size) = columns.size_hint();
         match size {
-            None => Err(ErrorCode::EmptyData("Can't concat empty columns")),
+            Some(0) => Err(ErrorCode::EmptyData("Can't concat empty columns")),
             Some(1) => Ok(columns.next().unwrap()),
```

Comment:
We should remove the `size_hint`, because it must not be trusted.

Reply:
Yeah